File: Baekjoon/Sorting/[15702]중간고사 채점/solution.py
N, M = map(int, input().split())

info = [[]for i in range(M)]
point = list(map(int, input().split()))

id, grade = (0, 0)

for i in range(M):
    row = input().split()

    id = int(row[0])
    grade = 0

    for j in range(N):
        if(row[j+1] == 'O'):
            grade += point[j]

    info[i] = [grade, id]

info.sort(key=lambda x: (-x[0], x[1]))

print(info[0][1], info[0][0])

File: Baekjoon/Sorting/[15702]중간고사 채점/solution.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

#define endl "\n"

typedef pair<int, int> pii;

vector<pii> info;
vector<int> point;
int N, M, id, grade;

int cmp(const pii a, const pii b);

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  cin >> N >> M;

  point = vector<int>(N);
  info = vector<pii>(M);

  for (int i = 0; i < N; i++) {
    cin >> point[i];
  }

  for (int i = 0; i < M; i++) {
    char tmp;
    grade = 0;

    cin >> id;
    for (int j = 0; j < N; j++) {
      cin >> tmp;

      if (tmp == 'O') grade += point[j];
    }

    info[i] = {grade, id};
  }

  sort(info.begin(), info.end(), cmp);

  cout << info[0].second << " " << info[0].first << endl;

  return 0;
}

int cmp(const pii a, const pii b) {
  if (a.first != b.first) {
    return a.first > b.first;
  } else {
    return a.second < b.second;
  }
}